code/modules/recycling/sortingmachinery.dm

```diff
@@ -409,11 +409,12 @@
 	. = ..()
 	if(.)
 		return
-	if(action != "change")
-		return
-	if(round(text2num(params["index"])) == currTag)
-		return
-	currTag = round(text2num(params["index"]))
+	switch(action)
+		if("change")
+			var/new_tag = round(text2num(params["index"]))
```

Comment:
Can you check that this is a valid index?

Reply:
Done.